src/coreclr/vm/jitinterface.cpp

```diff
@@ -5772,7 +5772,7 @@ bool __stdcall TrackAllocationsEnabled()
 }
 
 /***********************************************************************/
-CorInfoHelpFunc CEEInfo::getNewHelper(CORINFO_RESOLVED_TOKEN * pResolvedToken, CORINFO_METHOD_HANDLE callerHandle, bool * pHasSideEffects)
+CorInfoHelpFunc CEEInfo::getNewHelper(CORINFO_CLASS_HANDLE classHandle, CORINFO_METHOD_HANDLE callerHandle, bool* pHasSideEffects)
```

Comment:
Can `callerHandle` be deleted? It looks unused.

Reply:
It's used for diagnostics by crossgen2 – I initially actually was deleting it, until I ran into that use.

Reply:
We can replace the use by `methodBeingCompiled` -- it will have slightly different behavior when inlining, but maybe that's ok?

Reply:
Actually it shouldn't have any visible behavior difference since if R2R throws the exception during inlining it will just result in the inlining attempt being aborted, and the exception won't be surfaced to the user. So let me make this change.

Reply:
> We can replace the use by methodBeingCompiled -- it will have slightly different behavior when inlining, but maybe that's ok?

I do not think it will have different observable behavior. We catch and ignore exceptions during inlining, so the only place where this error can show up is when we fail during compilation of the top level method. When we fail during compilation of the top-level method, we are going to print its name in front of the error message. It is not necessary to duplicate the name of the method that failed to compile in the error message.